api/src/main/java/org/openmrs/module/htmlformentryui/element/ObsFromFragmentElement.java

```diff
@@ -67,10 +73,21 @@ public ObsFromFragmentElement(Map<String, String> parameters, UiUtils uiUtils) {
 		StringBuilder fragmentUrlBuilder = new StringBuilder();
 		fragmentUrlBuilder.append(fragment);
 		fragmentUrlBuilder.append("?");
-		fragmentUrlBuilder.append(parameters.get("fragmentParams"));
+		fragmentUrlBuilder.append(parameters.get("fragmentParams")); 
 		try {
-			fragmentParams = UiIncludeTagHandler.paramsToMap(fragmentUrlBuilder.toString());
-							
+			Map<String, Object> params = UiIncludeTagHandler.paramsToMap(fragmentUrlBuilder.toString());
+			// evaluate any velocity expressions
+			for (Entry<String, Object> entry : params.entrySet()) {
```

Comment:
I'm thinking it would make more sense for this block to live in the "parseFragmentParameters" block? (But open to arguments as to why it is better here)

Reply:
I guess this block was removed, in the latest commit. Maybe I needed to squash my commits into one